src/components/Tokens/TokenDetails/index.tsx

```diff
@@ -117,17 +118,27 @@ export default function TokenDetails({
     [urlAddress]
   )
 
-  const { chainId: connectedChainId } = useWeb3React()
+  const { account, chainId: connectedChainId } = useWeb3React()
   const pageChainId = supportedChainIdFromGQLChain(chain)
   const tokenQueryData = tokenQuery.token
-  const crossChainMap = useMemo(
-    () =>
-      tokenQueryData?.project?.tokens.reduce((map, current) => {
-        if (current) map[current.chain] = current.address
-        return map
-      }, {} as { [key: string]: string | undefined }) ?? {},
-    [tokenQueryData]
-  )
+  const { data: balanceQuery } = useCachedPortfolioBalancesQuery({ account })
+  const multiChainMap = useMemo(() => {
+    const tokenBalances = balanceQuery?.portfolios?.[0].tokenBalances
+    const tokensAcrossChains = tokenQueryData?.project?.tokens
+    if (!tokensAcrossChains) return {}
+    return tokensAcrossChains.reduce((map, current) => {
+      if (current) {
+        if (!map[current.chain]) {
+          map[current.chain] = {}
+        }
+        map[current.chain].address = current.address
+        map[current.chain].balance = tokenBalances?.find((tokenBalance) => tokenBalance.token?.id == current.id) as
+          | TokenBalance
+          | undefined
```

Comment:
Does this need to be casted?

Reply:
0xBa5c04Cdf4771ac6Da3f48fc368314a00DA064B7